### DISCUSSION TITLE
Block route in navigation

### DISCUSSION THREAD

**Discussion by hup333** [osmandapp/OsmAnd#20588] [Ideas]
Device : cancunf
Brand : motorola
Manufacturer : motorola
Model : moto g54 5G
Product : cancunf_g_sysen
Build : U1TDS34.94-12-7-6
Version : 14
App Version : OsmAnd+
Apk Version : 4.8.4 4804

Suggestion

If the road is blocked, or the track/path dosnt exist
An option to click on it avoid, to find a new route around

Road blocked for construction, no traffic information
A track is gone do to rain on a mountain
Or rocks has fallen
A tree or something blocking the road
Or flooding has flooded the road or tracks
Snow

**Comment by DmitryAlexei** [comment]
Have you tried avoid road option? For more information see
https://osmand.net/docs/user/map/map-context-menu#avoid-road

**Reply by hup333** [reply]
My bad
I have forgot about that option

Would be nice if the option was faster to acces, instead of having to dig into an menu
Click on the road, avoid in navigation (The route or track, are now avoided in the current navigation)

**Reply by ZeljkoBG** [reply]
Your objection applies to many options in the app. The guys know, they work, but obviously they don't have a project manager. Everything is 'scattered' in Settings. You need a college degree to set up an app.

**Reply by hup333** [reply]
Magic earth do it pretty easy
Just click avoid road, and select distance
Done
From the home screen, just Swype up
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/bf8b954d-6075-40c6-966a-12b1b5a54575"  width="300" height="auto"/> </td>
  </tr>
</table>

**Reply by mariush444** [reply]
@hup333 but it is as you wrote: click on the road then avoid the road in action botom menu. You can move action to the top.

**Reply by hup333** [reply]
From the website

Avoid road
Using Map context menu, you can add a specific road to the list of non-used for routing roads. That will affect [Route preparation](https://osmand.net/docs/user/navigation/setup/route-navigation).
To delete previosly selected roads from [Avoid roads list](https://osmand.net/docs/user/navigation/routing/osmand-routing#avoid-roads-menu) tap a sign, choose Remove (Android) or Dismiss (iOS).


Permanent avoid
I wasn't interested in that

note
Avoid roads function is global and affects all [Routing types](https://osmand.net/docs/user/navigation/routing/routing-types) (except [Online routing](https://osmand.net/docs/user/navigation/routing/online-routing)).
